From be3053b7bf92d680140f84ce65e982d75d6d996a Mon Sep 17 00:00:00 2001
From: Ma Jian <majian@jidemail.com>
Date: Wed, 6 Jul 2016 18:31:55 +0800
Subject: [PATCH 10/31] Fix build break on x86_64

When TARGET_ARCH=x86_64, the LOCAL_SRC_FILES will be wrongly set to
codegen.cpp.arm,  but the android make will 'smartly' remove the suffix
".arm" and try to compile the codegen.cpp without the libenc dependency.

NO_REF_TASK
Tested: local build

Change-Id: I25df0692886fd8b7c56d467d3b6f674919b09b56
---
 libpixelflinger/tests/codegen/Android.mk  | 4 ++--
 libpixelflinger/tests/codegen/codegen.cpp | 7 -------
 2 files changed, 2 insertions(+), 9 deletions(-)

diff --git a/libpixelflinger/tests/codegen/Android.mk b/libpixelflinger/tests/codegen/Android.mk
index 034e36fbb3..cd7ad8092d 100644
--- a/libpixelflinger/tests/codegen/Android.mk
+++ b/libpixelflinger/tests/codegen/Android.mk
@@ -1,7 +1,7 @@
 LOCAL_PATH:= $(call my-dir)
 include $(CLEAR_VARS)
 
-ifeq ($(TARGET_ARCH),x86)
+ifneq ($(filter x86%,$(TARGET_ARCH)),)
 LOCAL_SRC_FILES:= \
     codegen.cpp
 else
@@ -16,7 +16,7 @@ LOCAL_SHARED_LIBRARIES := \
 LOCAL_C_INCLUDES := \
 	$(LOCAL_PATH)/../..
 
-ifeq ($(TARGET_ARCH),x86)
+ifneq ($(filter x86%,$(TARGET_ARCH)),)
 LOCAL_STATIC_LIBRARIES := libenc
 endif
 
diff --git a/libpixelflinger/tests/codegen/codegen.cpp b/libpixelflinger/tests/codegen/codegen.cpp
index 47fa070d2b..b27a65342a 100644
--- a/libpixelflinger/tests/codegen/codegen.cpp
+++ b/libpixelflinger/tests/codegen/codegen.cpp
@@ -47,7 +47,6 @@ public:
     const AssemblyKey<needs_t>& key() const { return mKey; }
 };
 
-#if ANDROID_ARM_CODEGEN
 static void ggl_test_codegen(uint32_t n, uint32_t p, uint32_t t0, uint32_t t1)
 {
     GGLContext* c;
@@ -90,12 +89,6 @@ static void ggl_test_codegen(uint32_t n, uint32_t p, uint32_t t0, uint32_t t1)
 #endif
     gglUninit(c);
 }
-#else
-static void ggl_test_codegen(uint32_t, uint32_t, uint32_t, uint32_t) {
-    printf("This test runs only on ARM, Arm64 or MIPS\n");
-}
-#endif
-}
 
 int main(int argc, char** argv)
 {
-- 
2.17.1

